proposals/NNNN-experiment-tracking/README.md

```diff
@@ -0,0 +1,509 @@
+widespread popularity makes MLFlow SDK compatibility essential for Kubeflow's experiment tracking solution to gain
+traction among the AI/ML community. The MLFlow SDK's extensible
+[plugin architecture](https://mlflow.org/docs/latest/ml/plugins/#storage-plugins) allows developers to swap out the
+underlying backend storage while maintaining the familiar MLFlow API surface.
+
+To enable seamless integration, we will implement two key MLFlow plugins that connect to the Model Registry backend:
+
+1. **Tracking Store Plugin**: This plugin will handle experiment and run metadata, metrics, parameters, and tags. It
+   will map MLFlow's tracking concepts to the Model Registry's domain models (Experiments, Runs, Users, Metrics, etc.)
+   while excluding tracing functionality for the initial implementation.
+
+2. **Artifact Repository Plugin**: This plugin will manage artifact storage and retrieval, leveraging the Model
+   Registry's artifact management capabilities to store model files, datasets, and other ML artifacts.
+
+With these plugins in place, users will be able to use the MLFlow SDK with minimal changes to their existing code. They
+simply need to install an additional Python package and specify the Model Registry API as their tracking URI. This
+approach enables users to leverage MLFlow's powerful
+[automatic logging](https://mlflow.org/docs/latest/ml/tracking/autolog/) functionality for popular ML frameworks
+(TensorFlow, PyTorch, scikit-learn, etc.) while storing all data in the centralized Kubeflow Model Registry.
+
+**Future Enhancements**: In subsequent phases, we may extend the plugin architecture to include:
+
+- **Tracing Support**: Adding tracing capabilities akin to [MLflow](https://mlflow.org/docs/latest/genai/tracing/) would
+  require new domain models in Model Registry.
+- **Model Registry Store**: Further development could expose the Model Registry model registration functionality in the
+  MLFlow SDK.
+
+The primary objective remains enabling users to leverage MLFlow's mature ecosystem and familiar APIs while benefiting
+from Kubeflow's centralized experiment tracking infrastructure.
+
+#### Kubeflow SDK Implementation
+
+Building on the MLFlow Model Registry plugin, we can enhance the Kubeflow SDK to provide a native Kubeflow-centric
+experience. The initial implementation would focus on simplifying setup by automatically configuring the MLFlow Model
+Registry plugin and creating convenient wrapper functions around the supported MLFlow SDK APIs. This integration should
+feel cohesive with the existing Model Registry SDK functionality for registering models.
+
+Enhancements could include:
+
+- **Authentication**: Provide helpers for handling authentication with the Model Registry service, such as generating
+  Kubernetes tokens.
+- **Integrated UI Access**: Provide direct links to the Model Registry UI from within the Kubeflow SDK, making it easier
```

Comment:
This seems to suggest that the experiments UI will be part of the model Registry UI - is that correct? Or is the intention for the Experiments UI to be a standalone component?

Reply:
Yes, expanding the MR UI to include experiments was the intent. Thanks for pointing out that this wasn't clear. My latest push adds this clarification.